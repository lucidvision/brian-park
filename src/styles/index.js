export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
