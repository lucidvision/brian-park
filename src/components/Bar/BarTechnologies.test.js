import React from 'react'
import { shallow } from 'enzyme'
import BarTechnologies from './BarTechnologies'
import { colors } from 'styles'

const icons = [
  {
    className: 'fab fa-html5',
    color: colors.white,
    label: 'HTML5'
  },
  {
    className: 'fab fa-css3-alt',
    color: colors.white,
    label: 'CSS3'
  },
  {
    className: 'fab fa-js-square',
    color: colors.white,
    label: 'ES6'
  },
  {
    className: 'fab fa-react',
    color: colors.white,
    label: 'React'
  },
  {
    className: 'fab fa-react',
    color: colors.white,
    label: 'React-Native'
  },
  {
    className: 'fab fa-node-js',
    color: colors.white,
    label: 'Node'
  }
]

describe('BarTechnologies', () => {
  const barTechnologies = shallow(<BarTechnologies />)

  it('renders correctly', () => {
    expect(barTechnologies).toMatchSnapshot()
  })

  describe('Technology Icons', () => {
    it('renders 6 icons', () => {
      expect(barTechnologies.find('IconLabel')).toHaveLength(6)
    })

    icons.forEach((props, index) => {
      it(`renders the ${props.label} icon`, () => {
        expect(barTechnologies.find('IconLabel').at(index).props()).toEqual(props)
      })
    })
  })
})
