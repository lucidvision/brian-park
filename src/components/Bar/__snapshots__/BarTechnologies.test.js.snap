// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarTechnologies renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarTechnologies />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-html5"
                              color="#fff"
                              label="HTML5"
                    />
          </styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-css3-alt"
                              color="#fff"
                              label="CSS3"
                    />
          </styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-js-square"
                              color="#fff"
                              label="ES6"
                    />
          </styled.div>
</styled.div>,
        <styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-react"
                              color="#fff"
                              label="React"
                    />
          </styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-react"
                              color="#fff"
                              label="React-Native"
                    />
          </styled.div>
          <styled.div>
                    <IconLabel
                              className="fab fa-node-js"
                              color="#fff"
                              label="Node"
                    />
          </styled.div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <IconLabel
                            className="fab fa-html5"
                            color="#fff"
                            label="HTML5"
              />
</styled.div>,
            <styled.div>
              <IconLabel
                            className="fab fa-css3-alt"
                            color="#fff"
                            label="CSS3"
              />
</styled.div>,
            <styled.div>
              <IconLabel
                            className="fab fa-js-square"
                            color="#fff"
                            label="ES6"
              />
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-html5"
                color="#fff"
                label="HTML5"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-html5",
                "color": "#fff",
                "label": "HTML5",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-css3-alt"
                color="#fff"
                label="CSS3"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-css3-alt",
                "color": "#fff",
                "label": "CSS3",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-js-square"
                color="#fff"
                label="ES6"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-js-square",
                "color": "#fff",
                "label": "ES6",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <IconLabel
                            className="fab fa-react"
                            color="#fff"
                            label="React"
              />
</styled.div>,
            <styled.div>
              <IconLabel
                            className="fab fa-react"
                            color="#fff"
                            label="React-Native"
              />
</styled.div>,
            <styled.div>
              <IconLabel
                            className="fab fa-node-js"
                            color="#fff"
                            label="Node"
              />
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-react"
                color="#fff"
                label="React"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-react",
                "color": "#fff",
                "label": "React",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-react"
                color="#fff"
                label="React-Native"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-react",
                "color": "#fff",
                "label": "React-Native",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "children": <IconLabel
                className="fab fa-node-js"
                color="#fff"
                label="Node"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fab fa-node-js",
                "color": "#fff",
                "label": "Node",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-html5"
                                    color="#fff"
                                    label="HTML5"
                        />
            </styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-css3-alt"
                                    color="#fff"
                                    label="CSS3"
                        />
            </styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-js-square"
                                    color="#fff"
                                    label="ES6"
                        />
            </styled.div>
</styled.div>,
          <styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-react"
                                    color="#fff"
                                    label="React"
                        />
            </styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-react"
                                    color="#fff"
                                    label="React-Native"
                        />
            </styled.div>
            <styled.div>
                        <IconLabel
                                    className="fab fa-node-js"
                                    color="#fff"
                                    label="Node"
                        />
            </styled.div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <IconLabel
                                className="fab fa-html5"
                                color="#fff"
                                label="HTML5"
                />
</styled.div>,
              <styled.div>
                <IconLabel
                                className="fab fa-css3-alt"
                                color="#fff"
                                label="CSS3"
                />
</styled.div>,
              <styled.div>
                <IconLabel
                                className="fab fa-js-square"
                                color="#fff"
                                label="ES6"
                />
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-html5"
                  color="#fff"
                  label="HTML5"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-html5",
                  "color": "#fff",
                  "label": "HTML5",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-css3-alt"
                  color="#fff"
                  label="CSS3"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-css3-alt",
                  "color": "#fff",
                  "label": "CSS3",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-js-square"
                  color="#fff"
                  label="ES6"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-js-square",
                  "color": "#fff",
                  "label": "ES6",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <IconLabel
                                className="fab fa-react"
                                color="#fff"
                                label="React"
                />
</styled.div>,
              <styled.div>
                <IconLabel
                                className="fab fa-react"
                                color="#fff"
                                label="React-Native"
                />
</styled.div>,
              <styled.div>
                <IconLabel
                                className="fab fa-node-js"
                                color="#fff"
                                label="Node"
                />
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-react"
                  color="#fff"
                  label="React"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-react",
                  "color": "#fff",
                  "label": "React",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-react"
                  color="#fff"
                  label="React-Native"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-react",
                  "color": "#fff",
                  "label": "React-Native",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "children": <IconLabel
                  className="fab fa-node-js"
                  color="#fff"
                  label="Node"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fab fa-node-js",
                  "color": "#fff",
                  "label": "Node",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
